core/src/main/java/io/questdb/cutlass/line/tcp/StaticChallengeResponseMatcher.java

```diff
@@ -0,0 +1,65 @@
+/*******************************************************************************
+ *     ___                  _   ____  ____
+ *    / _ \ _   _  ___  ___| |_|  _ \| __ )
+ *   | | | | | | |/ _ \/ __| __| | | |  _ \
+ *   | |_| | |_| |  __/\__ \ |_| |_| | |_) |
+ *    \__\_\\__,_|\___||___/\__|____/|____/
+ *
+ *  Copyright (c) 2014-2019 Appsicle
+ *  Copyright (c) 2019-2023 QuestDB
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ *
+ ******************************************************************************/
+
+package io.questdb.cutlass.line.tcp;
+
+import io.questdb.cutlass.auth.AuthUtils;
+import io.questdb.cutlass.auth.ChallengeResponseMatcher;
+import io.questdb.log.Log;
+import io.questdb.log.LogFactory;
+import io.questdb.std.CharSequenceObjHashMap;
+import io.questdb.std.Chars;
+
+import java.nio.ByteBuffer;
+import java.security.InvalidKeyException;
+import java.security.PublicKey;
+import java.security.SignatureException;
+
+public class StaticChallengeResponseMatcher implements ChallengeResponseMatcher {
+    private static final Log LOG = LogFactory.getLog(StaticChallengeResponseMatcher.class);
+    private final CharSequenceObjHashMap<PublicKey> publicKeyByKeyId;
+    private final ByteBuffer signatureBuffer = ByteBuffer.allocate(72);
+
+    public StaticChallengeResponseMatcher(CharSequenceObjHashMap<PublicKey> authDb) {
+        this.publicKeyByKeyId = authDb;
+    }
+
+    @Override
```

Comment:
There's a small log-messsage side effect of not checking if publickKey was found . 
Previously it was :
`authentication failed, unknown key id`
now it's 
`java.security.InvalidKeyException: Key must not be null`

Reply:
do you think we should log it to the INFO level? it could be too much:
```
2023-06-29T11:37:02.663168Z I i.q.c.l.t.a.EllipticCurveAuthenticator [22] authentication read key id [keyId=invalid]
2023-06-29T11:37:02.681722Z I i.q.c.l.t.StaticChallengeResponseMatcher authentication failed, unknown key [id=invalid]
2023-06-29T11:37:02.681781Z I i.q.c.l.t.a.EllipticCurveAuthenticator [22] authentication failed, signature was not verified
```
wdyt?

Reply:
I don't have a strong opinion but think that logging that key is missing/can't be found  is useful .